test/test_loaders.py

```diff
@@ -38,6 +40,27 @@ def test_base_url_override(openapi3_schema_url, url):
     assert endpoint.base_url == "http://example.com"
 
 
+def test_port_override(openapi3_schema_url):
+    schema = schemathesis.from_uri(openapi3_schema_url, port=8081)
+    endpoint = next(schema.get_all_endpoints())
+    assert endpoint.base_url == "http://127.0.0.1:8081/schema.yaml"
+
+
+def test_port_override_with_ipv6(openapi3_schema_url, simple_openapi, mocker):
+    resp = Response()
+    resp.status_code = 200
+    resp._content = json.dumps(simple_openapi).encode("utf-8")
+    mocker.patch("requests.get", return_value=resp)
+
+    url = URL(openapi3_schema_url)
+    parts = list(map(int, url.host.split(".")))
+    ipv6_host = "2002:{:02x}{:02x}:{:02x}{:02x}::".format(*parts)
+    ipv6 = url.with_host("[%s]" % ipv6_host)
+    schema = schemathesis.from_uri(ipv6, validate_schema=False, port=8081)
```

Comment:
It fails due to passing URL instance instead of a string. It should be str(ipv6) instead